tests/unit/dataactbroker/test_permissions.py

```diff
@@ -4,75 +4,299 @@
+    user_frec, other_frec = FRECFactory(cgac=user_cgac), FRECFactory(cgac=other_cgac)
+    user_editfabs = UserFactory(affiliations=[
+        UserAffiliation(frec=user_frec, permission_type_id=ALL_PERMISSION_TYPES_DICT['editfabs'])
+    ])
+    database.session.add_all([user_cgac, other_cgac, user_frec, other_frec, user_editfabs])
+    database.session.commit()
+
+    monkeypatch.setattr(permissions, 'g', Mock(user=user_editfabs))
+
+    # has permission level, but wrong agency
+    assert not permissions.current_user_can('reader', None, other_frec.frec_code)
+    assert not permissions.current_user_can('editfabs', None, other_frec.frec_code)
+
+    # has agency, but not permission level
+    assert not permissions.current_user_can('writer', None, user_frec.frec_code)
+    assert not permissions.current_user_can('submitter', None, user_frec.frec_code)
+    assert not permissions.current_user_can('fabs', None, user_frec.frec_code)
+
+    # right agency, right permission
+    assert permissions.current_user_can('reader', None, user_frec.frec_code)
+    assert permissions.current_user_can('editfabs', None, user_frec.frec_code)
+
     # wrong permission level, wrong agency, but superuser
-    user_three.website_admin = True
-    assert permissions.current_user_can('submitter', other_cgac.cgac_code, None)
+    user_editfabs.website_admin = True
+    assert permissions.current_user_can('fabs', None, other_frec.frec_code)
+
+
+def test_current_user_can_fabs_frec_fabs(database, monkeypatch, user_constants):
+    user_cgac, other_cgac = [CGACFactory() for _ in range(2)]
+    user_frec, other_frec = FRECFactory(cgac=user_cgac), FRECFactory(cgac=other_cgac)
+    user_fabs = UserFactory(affiliations=[
+        UserAffiliation(frec=user_frec, permission_type_id=ALL_PERMISSION_TYPES_DICT['fabs'])
+    ])
+    database.session.add_all([user_cgac, other_cgac, user_frec, other_frec, user_fabs])
+    database.session.commit()
+
+    monkeypatch.setattr(permissions, 'g', Mock(user=user_fabs))
+
+    # has permission level, but wrong agency
+    assert not permissions.current_user_can('reader', None, other_frec.frec_code)
+    assert not permissions.current_user_can('editfabs', None, other_frec.frec_code)
+    assert not permissions.current_user_can('fabs', None, other_frec.frec_code)
+
+    # has agency, but not permission level
+    assert not permissions.current_user_can('writer', None, user_frec.frec_code)
+    assert not permissions.current_user_can('submitter', None, user_frec.frec_code)
+
+    # right agency, right permission
+    assert permissions.current_user_can('reader', None, user_frec.frec_code)
+    assert permissions.current_user_can('editfabs', None, user_frec.frec_code)
+    assert permissions.current_user_can('fabs', None, user_frec.frec_code)
+
+    # wrong agency, but superuser
+    user_fabs.website_admin = True
+    assert permissions.current_user_can('fabs', None, other_frec.frec_code)
 
```

Comment:
Putting this here because there isn't anywhere to just put it for the overall file. Can we get a test for someone with multiple permissions? Both dabs and fabs affiliations and make sure one doesn't overwrite the other in any way?